internal/config/validation.go

```diff
@@ -96,6 +97,53 @@ func DiscardNativeOnly(c ClusterResources) error {
 	return nil
 }
 
+// validateConfig is meant to validate all the inter-dependencies of a parsed configuration.
+// In this case, we ensure that bfd echo is not enabled on a pool.
```

Comment:
nit: on a v6 pool?
also, this being called "validateConfig" but being dedicated to a specific dependency feels a bit odd, wdyt about adding another function specifically for the bfd echo and calling it from validateConfig?
it would make validateConfig seem silly (just calling another function) but for long term when we find more inter-dependencies it makes sense to keep this function clean